core/db_models/EEM_Base.model.php

```diff
@@ -5421,6 +5431,16 @@ protected function _get_column_value_with_table_alias_or_not($cols_n_values, $qu
             $value = $cols_n_values[ $qualified_column ];
         } elseif (isset($cols_n_values[ $regular_column ])) {
             $value = $cols_n_values[ $regular_column ];
+        } elseif (! empty($this->foreign_key_aliases)) {
+            // no PK?  ok check if there is a foreign key alias set for this table
+            // then check if that alias exists in the incoming data
+            // AND that the actual PK the $FK_alias represents matches the $qualified_column (full PK)
+            foreach ($this->foreign_key_aliases as $FK_alias => $PK_column) {
+                if ($PK_column === $qualified_column && isset($cols_n_values[ $FK_alias ])) {
+                    $value = $cols_n_values[ $FK_alias ];
+                    break;
+                }
+            }
         }
         return $value;
```

Comment:
this is essentially where the bug was...
The fully qualified primary key for the `EEM_WP_User` model is `WP_User.ID` but other models like `EEM_Event` specify that key like `Event.EVT_wp_user`.
So the incoming data might include a value for `Event.EVT_wp_user, but we were only checking for `WP_User.ID` or `ID` which of course would not exist.